### PR TITLE
feat(gsdk): unlock all calls and storage queries

### LINKED ISSUE #2306: Introduce manual dynamic::tx methods in gsdk
Mb introduce manual dynamic::tx method to put some enums inside? To avoid misprinting in strs

_Originally posted by @breathx in https://github.com/gear-tech/gear/pull/2280#discussion_r1118918902_

### PR DESCRIPTION
**Pull request by clearloop**
Resolves #2306 

- [x] generate call info
  - [x] use enums in dynamic::tx
- [x] generate storage info
  - [x] use enums in dynamic::storage
- [x] more docs
  - [x] parsing events
  - [x] executing transaction manually
  - [x] querying storage manually

## Notes

Users now can call any calls or storages with `gsdk` instead of our prebuilt APIs only

@gear-tech/dev

### CODE REVIEW

**Comment by breathx** [comment]
CI fails

File: Cargo.lock [hunk 1]
```diff
@@ -27,7 +27,7 @@ version = "0.19.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "a76fd60b23679b7d19bd066031410fb7e458ccc5e958eb5c325888ce4baedc97"
 dependencies = [
```
Comment (breathx) [inline Cargo.lock:29]: Fix lock pls, it dumps versions

File: gsdk/api-gen/src/main.rs [hunk 2]
```diff
@@ -68,21 +69,24 @@ fn main() -> Result<()> {
         panic!("Invalid metadata, doesn't even have enough bytes for the magic number.");
     }
 
-    // NOTE: [4..] here for removing the magic number.
+    // NOv TE: [4..] here for removing the magic number.
```
Comment (breathx) [inline gsdk/api-gen/src/main.rs:72]:
```suggestion
    // NOTE: [4..] here for removing the magic number.
```

File: gsdk/build.rs [hunk 3]
```diff
@@ -49,6 +49,8 @@ fn generate_api() -> Vec<u8> {
     ]
     .map(|(relative_path, pkg, features)| get_path(root, &profile, relative_path, pkg, features));
 
+    // rebuild mmmm.
```
Comment (breathx) [inline gsdk/build.rs:52]: xd 
```suggestion
```

File: gsdk/src/lib.rs [hunk 4]
```diff
@@ -84,6 +84,24 @@ impl Api {
...
+    ///
+    /// # Parse events from blocks
+    ///
+    /// ```ignore
+    /// use gsdk::metadata::Event;
+    ///
+    /// let api = Api::new(None).await?;
+    /// let mut blocks = api.blocks().await?;
+    ///
+    /// while let Some(Ok(events)) = blocks.next_events().await {
```
Comment (breathx) [inline gsdk/src/lib.rs:96]: btw, can't we cast events as root events just like here?

File: gsdk/src/signer/utils.rs [hunk 5]
```diff
@@ -33,4 +39,64 @@ impl Signer {
...
+    ///         Value::unnamed_variant("Id", [Value::from_bytes(dest.into())]),
+    ///         Value::u128(value),
+    ///     ];
+    ///     let in_block = signer.run_tx(BalancesCall::Transfer, args).await?;
+    /// }
+    ///
+    /// // The code above euqals to:
+    ///
+    /// {
+    ///    let in_block = signer.transfer(dest, value).await?;
```
Comment (breathx) [inline gsdk/src/signer/utils.rs:77]: I'm not pretty sure that it affects api positively. I'd go with
`signer.calls.transfer`
and
`signer.storage.number`
to split functionality and make it more readable + intuitive

File: gsdk/src/storage.rs [hunk 6]
```diff
@@ -49,6 +53,32 @@ use subxt::{
...
+    /// use gsdk::{Api, metadata::storage::SystemStorage};
+    ///
+    /// let api = Api::new(None);
+    ///
+    /// {
+    ///     let address = Api::storage(SystemStorage::Number);
+    ///     let bn = api.fetch_storage(address).await?;
+    /// }
+    ///
+    /// // The code above equals to:
```
Comment (breathx) [inline gsdk/src/storage.rs:75]: due to implemented function*

**Review by breathx — APPROVED** [review]

File: gsdk/src/lib.rs (hunk 4 above)
Comment (clearloop) [reply gsdk/src/lib.rs:96]: do you mean adding a new method `events`?

File: gsdk/src/signer/utils.rs (hunk 5 above)
Comment (clearloop) [reply gsdk/src/signer/utils.rs:77]: #2895

File: gsdk/src/lib.rs (hunk 4 above)
Comment (clearloop) [reply gsdk/src/lib.rs:96]: #2896

File: Cargo.lock (hunk 1 above)
Comment (clearloop) [reply Cargo.lock:29]: tried `cargo update` but nothing changed

**Review by shamilsan — APPROVED** [review]

File: Cargo.lock (hunk 1 above)
Comment (breathx) [reply Cargo.lock:29]: just merge master, copy-paste lock from master and run build - everything required will be bumped

**Pull request merged**